target/linux/lantiq/files-4.19/arch/mips/boot/dts/lantiq/grx390_dlink_dwr-966.dts

```diff
@@ -0,0 +1,288 @@
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+		};
+		rssi_3 {
+			label = "green:rssi_3";
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+		};
+		rssi_4 {
+			label = "green:rssi_2";
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+		};
+		rssi_5 {
+			label = "green:rssi_1";
+			gpios = <&gpio 18 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&localbus {
+	nand@1 {
+		compatible = "lantiq,nand-xway";
+		lantiq,cs = <1>;
+		bank-width = <2>;
+		reg = <1 0x0 0x2000000>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "uboot";
+				reg = <0x00000 0x100000>;
+				read-only;
+			};
+			partition@100000 {
+				label = "ubootconfigA";
+				reg = <0x100000 0x40000>;
+				read-only;
+			};
+			partition@140000 {
+				label = "ubootconfigB";
+				reg = <0x140000 0x40000>;
+				read-only;
+			};
+			partition@180000 {
+				label = "gphyfirmware";
+				read-only;
+				reg = <0x180000 0x40000>;
+			};
+			partition@01c0000 {
+				label = "ubi";
+				reg = <0x01c0000 0x7c00000>;
+			};
+			partition@7dc0000 {
+				label = "calibration";
+				reg = <0x7dc0000 0x0100000>;
+				read-only;
+			};
+			partition@7ec0000 {
+				label = "factory";
+				reg = <0x7ec0000 0x0100000>;
+				read-only;
+			};
+			partition@7fc0000 {
+				label = "res";
+				reg = <0x7fc0000 0x0040000>;
+				read-only;
+			};
+
+		};
+	};
+};
+
+&eth0 {
+	lan: interface@0 {
+		compatible = "lantiq,xrx200-pdi";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <0>;
+		mac-address = [ 00 11 22 33 44 55 ];
+		lantiq,switch;
+
+		ethernet@1 {
+			compatible = "lantiq,xrx200-pdi-port";
+			reg = <1>;
+			phy-mode = "gmii";
+			phy-handle = <&phy1>;
+		};
+		ethernet@2 {
+			compatible = "lantiq,xrx200-pdi-port";
+			reg = <2>;
+			phy-mode = "gmii";
+			phy-handle = <&phy2>;
+		};
+		ethernet@3 {
+			compatible = "lantiq,xrx200-pdi-port";
+			reg = <3>;
+			phy-mode = "gmii";
+			phy-handle = <&phy3>;
+		};
+		ethernet@4 {
+			compatible = "lantiq,xrx200-pdi-port";
+			reg = <4>;
+			phy-mode = "gmii";
+			phy-handle = <&phy4>;
+		};
+	};
+
+	mdio@0 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "lantiq,xrx200-mdio";
+		reg = <0>;
+
+		phy1: ethernet-phy@1 {
+			reg = <0x1>;
+			compatible = "lantiq,phy11g", "ethernet-phy-ieee802.3-c22";
+		};
+		phy2: ethernet-phy@2 {
+			reg = <0x2>;
+			compatible = "lantiq,phy11g", "ethernet-phy-ieee802.3-c22";
+		};
+		phy3: ethernet-phy@3 {
+			reg = <0x3>;
+			compatible = "lantiq,phy11g", "ethernet-phy-ieee802.3-c22";
+		};
+		phy4: ethernet-phy@4 {
+			reg = <0x4>;
+			compatible = "lantiq,phy11g", "ethernet-phy-ieee802.3-c22";
+		};
+	};
+};
+
+&gpio {
+	pinctrl-names = "default";
+	pinctrl-0 = <&state_default>;
+
+	state_default: pinmux {
+		mdio {
+			lantiq,groups = "mdio";
+			lantiq,function = "mdio";
+		};
+
+		nand_out {
+			lantiq,groups = "nand cle", "nand ale", "nand rd",
+						"nand d1",
```

Comment:
That indent does not really look helpful.

Reply:
fixed